model/user.go

```diff
@@ -0,0 +1,249 @@
+package model
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/vocdoni/blind-csp/types"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.vocdoni.io/dvote/log"
+)
+
+var (
+	ErrUserUnknown    = fmt.Errorf("user is unknown")
+	ErrUserDuplicated = fmt.Errorf("user is duplicated")
+)
+
+type UserRequest struct {
+	UserID   string `json:"userId"`
+	Handler  string `json:"handler"`
+	Service  string `json:"service"`
+	Mode     string `json:"mode"`
+	Data     string `json:"data"`
+	Consumed *bool  `json:"consumed,omitempty"`
+}
+
+type User struct {
+	ID         types.HexBytes `json:"userId" bson:"_id"`
+	ElectionID types.HexBytes `json:"electionId" bson:"electionId"`
+	Handler    string         `json:"handler" bson:"handler"`
+	Service    string         `json:"service" bson:"service"`
+	Mode       string         `json:"mode" bson:"mode"`
+	Data       string         `json:"data" bson:"data"`
+	Consumed   *bool          `json:"consumed" bson:"consumed"`
+}
+
+type UserStore interface {
+	Create(electionID types.HexBytes, handler HandlerConfig, userData string) (*User, error)
+	User(electionID types.HexBytes, userID types.HexBytes) (*User, error)
+	Update(electionID types.HexBytes, userID types.HexBytes, userR UserRequest) (*User, error)
+	Delete(electionID types.HexBytes, userID types.HexBytes) error
+	List(electionID types.HexBytes) (*[]User, error)
```

Comment:
To be consistent use Users, AddUser, etc. Also in this way we are following a same patter in the repository

Reply:
Not sure about this. Why `Users` ? AFAIU User() is a getter of the type User, so its correct.

Reply:
She means that some of these methods don't say "User" at all, like Create or List, so they're not consistent nor explicit. You'd likely want CreateUser, GetUser, UpdateUser, etc. Another problem with the current interface is that if later on you have a second similar interface, you won't be able to implement both, since names like `Update` will conflict.

Reply:
So, should the getter be User() or GetUser() ?

Reply:
I understand now, thanks @mvdan, yes it makes sense.

Let's call it User() instead of GetUser() @nigeon. The getter is explicit by itself, no need for the Get prefix.

Reply:
> She means that some of these methods don't say "User" at all, like Create or List, so they're not consistent nor explicit. You'd likely want CreateUser, GetUser, UpdateUser, etc. Another problem with the current interface is that if later on you have a second similar interface, you won't be able to implement both, since names like `Update` will conflict.

Yeah, exactly

Reply:
Ok!